player/scripting.c

```diff
@@ -85,10 +85,10 @@ static char *script_name_from_filename(void *talloc_ctx, const char *fname)
 
 static void run_script(struct mp_script_args *arg)
 {
-    char name[90];
-    snprintf(name, sizeof(name), "%s (%s)", arg->backend->name,
-             mpv_client_name(arg->client));
+    char *name = talloc_asprintf(NULL, "script/%s/%s", arg->backend->name,
+                                 mpv_client_name(arg->client));
```

Comment:
I'd go for `script/{client}-{backend}` instead so that the `arg->client` doesn't get fully truncated. `scripts/javascript/` is 20 bytes on it's own already.

Reply:
Just for the record, we also perpend `mpv/` to each name, if we want to support 15 char limit, we should probably remove `mpv/` prefix, for scripts probably use `%s/%s` only. I would like to keep lua, native, js prefix. Probably rename `javascript` to `js` . For all other constant string cases, I will rename it to fit.

Currently it looks like this.
```
29300	mpv/ao/wasapi	mpv.exe!audio_thread
38820	mpv/demux	mpv.exe!mp_cond_timedwait
38220	mpv/main	mpv.exe!mp_cond_timedwait
35612	mpv/script/lua/autocrop	mpv.exe!mp_cond_timedwait
26928	mpv/script/lua/console	mpv.exe!mp_cond_timedwait
40996	mpv/script/lua/stats	mpv.exe!mp_cond_timedwait
42016	mpv/script/lua/uosc	mpv.exe!mp_cond_timedwait
37740	mpv/script/lua/ytdl_hook	mpv.exe!mp_cond_timedwait
18260	mpv/terminal/input	mpv.exe!input_thread_fn
30468	mpv/vo	mpv.exe!mp_cond_timedwait
39028	mpv/window	win32u.dll!NtUserGetMessage
43164	mpv/worker	mpv.exe!mp_cond_timedwait
```

Reply:
> Just for the record, we also perpend mpv/ to each name

Hmmm... I guess we can ignore the limit for now. Lots of things are being truncated on glibc at the current state as well but it seems like no one really cared. If someone cares, he can pick it up later.

Reply:
I've updated. I'd like to have longer names, but at lest now it is an improvement to both consistency and length. I don't think anyone even notices the thread names, so anyway...

Reply:
Why allocate memory here? It's just going to get truncated anyway.

Reply:
> I'd like to have longer names

After learning about the "mpv/" prefix along with looking at many cases where it's getting truncated *already*, I no longer really care about enforcing that. So feel free to add "mpv/" back or use longer names. :+1: 

> Why allocate memory here? It's just going to get truncated anyway.

Previously everything was truncated to 80 bytes regardless of the platform. But after Kasper's recent change, only the linux branch gets truncated on `ERANGE`.

Reply:
> After learning about the "mpv/" prefix along with looking at many cases where it's getting truncated already, I no longer really care about enforcing that. So feel free to add "mpv/" back or use longer names. 👍

I hear you, but without much effort we can make it more "compatible" with a limit. FWIW, we could set the prefix on win32 only after mp_thread is merged and on Linux left it as is to save chars as much as possible.

I'm fine with current PR, the names are consistent and distinct for most part.

> Why allocate memory here? It's just going to get truncated anyway.

No need to truncate to arbitrary 80. On some platforms it will not be truncated later. Setting name is don once per thread, we can afford one allocation.